modules/core/src/shaderlib/shadow/shadow.js

```diff
@@ -135,19 +163,24 @@ function createShadowUniforms(opts = {}, context = {}) {
   const viewProjectionMatrices = [];
   const projectCenters = [];
 
-  for (let i = 0; i < opts.shadow_viewProjectionMatrices.length; i++) {
-    const viewProjectionMatrix = opts.shadow_viewProjectionMatrices[i]
+  for (let i = 0; i < opts.shadow_matrices.length; i++) {
+    const viewProjectionMatrix = _getMemoizedViewProjectionMatrix({
+      shadowMatrix: opts.shadow_matrices[i],
```

Comment:
The memoized function is still recomputing every time if you have more than one light.